web-client/src/views/FileHandlingHelpers/fileValidation.ts

```diff
@@ -0,0 +1,204 @@
+  '.jpg': 'a JPEG Image',
+  '.pdf': 'a PDF',
+  '.png': 'a PNG Image',
+};
+
+const getDisplayNameForFileExtension = (fileExtension: string) => {
+  if (Object.keys(fileExtensionDisplayNames).includes(fileExtension)) {
+    return fileExtensionDisplayNames[fileExtension];
+  }
+  return fileExtension;
+};
+
+const getWrongFileTypeMessage = (fileExtensions: string[]) => {
+  if (fileExtensions.length === 1) {
+    const displayName = getDisplayNameForFileExtension(fileExtensions[0]);
+    return `The file is not ${displayName}. Select ${displayName} file or resave the file as ${displayName}.`;
+  }
+  return `The file is not in a supported format (${fileExtensions.join(', ')}). Select a different file or resave it in a supported format.`;
+};
+
+export const validateFileSize = ({
+  file,
+  megabyteLimit,
+}: {
+  file: File;
+  megabyteLimit: number;
+}): FileValidationResponse => {
+  if (file.size > megabyteLimit * 1024 * 1024) {
+    return {
+      errorInformation: {
+        errorMessageToDisplay: `The file size is too big. The maximum file size is ${megabyteLimit}MB. Reduce the file size and try again.`,
+        errorType: ErrorTypes.FILE_TOO_BIG,
+      },
+      isValid: false,
+    };
+  }
+  return { isValid: true };
+};
+
+const validateCorrectFileType = ({
+  allowedFileExtensions,
+  file,
+}: {
+  file: File;
+  allowedFileExtensions: string[];
+}): FileValidationResponse => {
+  if (!allowedFileExtensions.includes(getFileExtension(file.name))) {
+    return {
+      errorInformation: {
+        errorMessageToDisplay: getWrongFileTypeMessage(allowedFileExtensions),
+        errorType: ErrorTypes.WRONG_FILE_TYPE,
+      },
+      isValid: false,
+    };
+  }
+  return { isValid: true };
+};
+
+export const validateFileOnSelect = async ({
+  allowedFileExtensions,
+  e,
+  megabyteLimit,
+  onError,
+  onSuccess,
+}: {
+  allowedFileExtensions: string[];
+  e: React.ChangeEvent<HTMLInputElement>;
+  megabyteLimit: number;
+  onSuccess: ({ selectedFile }: { selectedFile: File }) => void;
+  onError: ({
+    errorType,
+    messageToDisplay,
+    messageToLog,
+  }: {
+    messageToDisplay: string;
+    errorType?: ErrorTypes;
+    messageToLog?: string;
+  }) => void;
+}) => {
+  const target = e.target as HTMLInputElement;
+
+  if (!target || !target.files || target.files.length === 0) {
+    onError({
+      messageToDisplay: 'No file selected. Please upload a file.',
+    });
+    return;
+  }
+
+  const selectedFile = target.files[0];
+  const { errorInformation, isValid } = await validateFile({
+    allowedFileExtensions,
+    file: selectedFile,
+    megabyteLimit,
+  });
+  if (!isValid) {
+    onError({
+      errorType: errorInformation?.errorType,
+      messageToDisplay: errorInformation?.errorMessageToDisplay!,
+      messageToLog: errorInformation?.errorMessageToLog,
+    });
+    target.value = '';
+    return;
+  }
+  onSuccess({ selectedFile });
+};
+
+export const validateFile = async ({
```

Comment:
Is this function called anywhere apart from `validateFileOnSelect` in this file? In that case, do we necessarily need to export it and put it under test? Just a thought if we want to trim down the test suite for this file.

Reply:
Good point--I think you're correct! Thank you :)